pkg/csi/service/wcpguest/controller.go

```diff
@@ -154,6 +179,19 @@ func (c *controller) Init(config *cnsconfig.Config) error {
 		log.Errorf("failed to watch on path: %q. err=%v", cnsconfig.DefaultpvCSIProviderPath, err)
 		return err
 	}
+	// Go module to keep the metrics http server running all the time.
+	go func() {
+		csiInfo.WithLabelValues(version).Set(1)
+		for {
+			log.Info("Starting the http server to expose Prometheus metrics..")
+			http.Handle("/metrics", promhttp.Handler())
+			err = http.ListenAndServe(":2112", nil)
```

Comment:
Why are we using port "2112"? 
Prometheus is recommending to grab the free https://github.com/prometheus/prometheus/wiki/Default-port-allocations

> Feel free to grab the next free port number when developing your exporter, preferably before publicly announcing it. If you’re not ready to release yet, putting your username and WIP is fine.

Reply:
The link you pointed to is for introducing any new ports in Prometheus. We are not interested in that.

The port we are interested is the one on which we want to run the http server that exposes the metrices. We can decide on any port, I followed the go example in Prometheus doc for this - https://prometheus.io/docs/guides/go-application/. This uses 2112, but we are free to use any port as long as that port is not already used by something in CSI controller Pod. This also reminds me that I need to change the PVCSI yaml, which I will address in the next commit.